lib/active_model/serializer/attributes.rb

```diff
@@ -55,6 +55,9 @@ def attributes(*attrs)
         #     end
         def attribute(attr, options = {}, &block)
           key = options.fetch(:key, attr)
+          # replace ? at the end of an attribute, as JSON API disallows the use
+          # of ? at the end of keys.
+          key = key.to_s.gsub(/\?\z/, '').to_sym
```

Comment:
I'll be picky here but I usually tend to avoid using `gsub` when `sub` can be used: ` key = key.to_s.sub(/\?\z/, '').to_sym` :grin: